### DISCUSSION TITLE
How do I add a new alias?

### DISCUSSION THREAD

**Discussion by filimon-danopoulos-stratsys** [storybookjs/storybook#24822] [Help] [answered]
I have two directories, `src` that contains source code for components and stories and `storybook` that contains configuration, custom docblocks and custom helpers. I have an the alias configured as follows: 

```js
// tsconfig.json
"paths": {
  "@/*": [
    "./src/*"
  ],
  "$/*": [
    "./storybook/*"
  ]
}
```
```js
// webpack.config.js
resolve: {
  alias: {
    '@': path.resolve(process.cwd(), 'src'),
    $: path.resolve(process.cwd(), 'storybook')    
  }
}
```
```js
// storybook/main.js

webpackFinal: async (config, { configType }) => {
  config.resolve.alias['@'] = webpackConfig.resolve.alias['@']
  config.resolve.alias['$'] = webpackConfig.resolve.alias['$']
}
```
The alias `@` works but the `$` does not. Are there known limitations for aliases?

**Comment by valentinpalkovic** [comment]
I think you have forgotten the `.` in front of the storybook folder:

```tsx
// webpack.config.js
resolve: {
  alias: {
    '@': path.resolve(process.cwd(), 'src'),
    // adding dot
    $: path.resolve(process.cwd(), '.storybook')    
  }
}

**Reply by filimon-danopoulos-stratsys** [reply]
Sorry forgot to add that I am not using the default `.storybook` I have a folder called `storybook`.

**Reply by valentinpalkovic** [reply]
But your tsconfig path alias maps to `.storybook`. Is that intended?

```tsx
// tsconfig.json
"paths": {
  "@/*": [
    "./src/*"
  ],
  "$/*": [
    "./storybook/*"
  ]
}
```

**Reply by valentinpalkovic** [reply]
Also, I would rather recommend to use the [tsconfig-paths-webpack-plugin](https://www.npmjs.com/package/tsconfig-paths-webpack-plugin) to reuse the mapping of your tsconfig. The plugin is automatically configured for Angular and Nextjs. For the other frameworks, you would have to add it by yourself.

**Reply by filimon-danopoulos-stratsys** [reply]
It maps to `./storybook/*` not `.storybook/*`. Will look into that plugin thanks and see if it helps thanks.

**Reply by valentinpalkovic** [reply]
oh please apologise! It's still early in the morning xD

**Reply by valentinpalkovic** [reply, marked as answer]
I think it is not working because the dollar sign has a specific meaning in Webpack's resolve algorithm: https://webpack.js.org/configuration/resolve/#resolvealias.

**Reply by filimon-danopoulos-stratsys** [reply]
That could probably be the case, will try another symbol and see if it works.

**Reply by valentinpalkovic** [reply]
Let me know if that worked!

**Reply by filimon-danopoulos-stratsys** [reply]
Changed to `#` and it works as intended, thanks.
